### ISSUE TITLE
mimetic interpolation for edge- and face- fields

### ISSUE THREAD

**Issue by BenWibking** [parthenon-hpc-lab/parthenon#970]
There is an elegant method described here for linear, conservative interpolation of edge- and face-centered fields that preserves curl-free- or divergence-free-ness of the fields after interpolation: https://ui.adsabs.harvard.edu/abs/2015JCoPh.302...21P/abstract

**Comment by lroberts36** [comment]
I think it will be important to implement something like that once we implement non-Cartesian coordinate systems and working in terms of p-forms certainly makes things simpler. The prolongation operators that are currently in Parthenon are conservative for all types of fields in Cartesian coordinate systems, though.

**Comment by Yurlungur** [comment]
Yeah, we've discussed this in the hallway before and I am in favor. The differential geometry picture helps one handle lots of the complexity.

**Comment by BenWibking** [comment]
> I think it will be important to implement something like that once we implement non-Cartesian coordinate systems and working in terms of p-forms certainly makes things simpler. The prolongation operators that are currently in Parthenon are conservative for all types of fields in Cartesian coordinate systems, though.

Ah, neat. I didn't realize that was already there for edge and face fields in Cartesian coordinates.

I have been thinking about this in the context of edge-centered vector potentials, since this seems to avoid various issues with evolving the face-centered fields for CT.
